codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithyrust/generator/RustAwsSdkShimGenerator.java

```diff
@@ -0,0 +1,1739 @@
+            >,
+        ) -> $sdkCrate:L::types::$rustStructureName:L {
+            $sdkCrate:L::types::$rustStructureName:L::builder()
+                  $fluentMemberSetters:L
+                  .build()
+                  $unwrapIfNeeded:L
+        }
+        """,
+        variables
+      )
+    );
+  }
+
+  private RustFile enumConversionModule(
+    final ServiceShape service,
+    final Shape enumShape
+  ) {
+    Path path = Path.of(
+      "src",
+      "conversions",
+      toSnakeCase(enumShape.getId().getName()) + ".rs"
+    );
+
+    return new RustFile(
+      path,
+      TokenTree.of(
+        enumToDafnyFunction(enumShape),
+        enumFromDafnyFunction(enumShape)
+      )
+    );
+  }
+
+  private TokenTree enumToDafnyFunction(final Shape enumShape) {
+    String enumName = enumShape.getId().getName();
+    String rustEnumName = toPascalCase(enumName);
+    String sdkId = service
+      .expectTrait(ServiceTrait.class)
+      .getSdkId()
+      .toLowerCase();
+    String dafnyTypesModuleName =
+      "software::amazon::cryptography::services::%s::internaldafny::types".formatted(
+          sdkId
+        );
+    Map<String, String> variables = Map.of(
+      "sdkCrate",
+      "aws_sdk_" + sdkId,
+      "enumName",
+      enumName,
+      "rustEnumName",
+      rustEnumName,
+      "dafnyTypesModuleName",
+      dafnyTypesModuleName
+    );
+
+    String sdkTypeName = evalTemplate(
+      "$sdkCrate:L::types::$rustEnumName:L",
+      variables
+    );
+
+    var prelude = TokenTree.of(
+      evalTemplate(
+        """
+        #[allow(dead_code)]
+
+        pub fn to_dafny(
+            value: $sdkCrate:L::types::$rustEnumName:L,
+        ) -> ::std::rc::Rc<crate::r#$dafnyTypesModuleName:L::$enumName:L>{
+            ::std::rc::Rc::new(match value {
+
+        """,
+        variables
+      )
+    );
+
+    var branches = TokenTree
+      .of(
+        enumShape
+          .expectTrait(EnumTrait.class)
+          .getValues()
+          .stream()
+          .map(e ->
+            TokenTree.of(
+              sdkTypeName +
+              "::" +
+              rustEnumName(e) +
+              " => crate::r#" +
+              dafnyTypesModuleName +
+              "::" +
+              enumName +
+              "::" +
+              dafnyEnumName(e) +
+              " {},"
+            )
+          )
+      )
+      .lineSeparated();
+    final var postlude = TokenTree.of(
+      """
+
+              // TODO: This should not be a panic, but the Dafny image of the enum shape doesn't have an Unknown variant of any kind,
+              // so there's no way to succeed.
+              // See https://github.com/smithy-lang/smithy-dafny/issues/476.
+              // This could be handled more cleanly if conversion functions returned Results,
+              // but that would be a large and disruptive change to the overall code flow.
+              _ => panic!("Unknown enum variant: {}", value),
```

Comment:
Is there a reason to put this TODO in the generated code instead of in the generating code?

Reply:
Good point, I'll move it to the code generator instead.